sphinx/builders/linkcheck.py

```diff
@@ -120,12 +120,16 @@ def init(self):
 
     def check_thread(self):
         # type: () -> None
-        kwargs = {}
+        kwargs = {
+            'allow_redirects': True,
+            'headers': {
+                'Accept': ('text/html,application/xhtml+xml,'
+                           'application/xml;q=0.9,*/*;q=0.8')
```

Comment:
I don't understand why `application/xml` is needed. Can Sphinx recognize XML document?

Reply:
I'd have to test it again without `application/xml` to verify, but in my opinion even without it, it would still solve the issue I had with `crates.io`.

The reason I added the header like this is to mimic a real browser, 1:1. So I just took whatever my own Firefox is sending.

Since the link check verifies whether the external links in the docs are "not dead" and still accessible for humans reading the docs and clicking the links, displaying them in their browsers, I think it is a correct approach. If browsers are generally fine with XML and prefer it over other media types, then I think it should be there.

Reply:
The linkcheck builder uses `HTMLParser` to parse the response. I'm okay if it can handle any XML data. To send `Accept: application/xml` means the client can accept it as a data format. But I don't know our builder supports it.

Reply:
Note: Even `application/xml` not present, `creates.io` returns its contents:
```
$ curl -H 'Accept: text/html,application/xhtml+xml;q=0.9,*/*;q=0.8' -i https://crates.io/crates/dredd-hooks
HTTP/1.1 200 OK
...
```